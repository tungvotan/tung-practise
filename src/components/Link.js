// @flow
import * as React from 'react'
import PropTypes from 'prop-types'

type LinkPropsType = {
  active: boolean,
  children: React.Node,
  onClick: () => void
}

const Link = ({ active, children, onClick } : LinkPropsType) => (
    <button
       onClick={onClick}
       disabled={active}
       style={{
           marginLeft: '4px',
       }}
    >
      {children}
    </button>
)

export default Link
