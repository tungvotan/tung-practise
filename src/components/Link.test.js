import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Link from './Link';

const setup = (setupProps = {}) => {
  const wrapper = shallow(<Link/>);

  return {
    wrapper
  };
};

describe('Link component', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
