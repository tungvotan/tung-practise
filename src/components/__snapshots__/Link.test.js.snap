// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders without crashing 1`] = `
<button
  style={
    Object {
      "marginLeft": "4px",
    }
  }
/>
`;
